### PR TITLE
[ui] Show one asset run tag or “4 assets”, hover action for lineage

### PR DESCRIPTION
**Pull request by bengotow**
## Summary & Motivation

This PR changes the asset tags that appear in the run table and on the run details page:

1) Rather than show 1-3 individual assets or a "6 assets" link in the run table or run details header, we now show either a single asset or "6 assets". This means the rendering is more compact in the 2-3 asset case since the tags are not rendered individually.

2) Instead of being directly clickable, hovering over the assets gives you the option to jump to a lineage view OR open the list of assets. For >1 asset, the downstream lineage option goes to the global asset graph with the "assetA*, assetB*" query. It's actually pretty slick.

Right now in Dagit we don't have a "default behavior" for clicking tags that have tag actions -- you have to choose an action. I could see keeping the default "link" behavior we had before though...

Before:
image
image
image

After:
image
image
image

Also verified with multi-component asset keys:
image
image

## How I Tested These Changes

I viewed the runs list and run details for a wide range of runs.

### CODE REVIEW

File: js_modules/dagster-ui/packages/ui-core/src/runs/RunConfigDialog.tsx [hunk 1]
```diff
@@ -39,99 +36,8 @@ import {doneStatuses} from './RunStatuses';
 import {RunTags} from './RunTags';
 import {RunsQueryRefetchContext} from './RunUtils';
 import {TerminationDialog} from './TerminationDialog';
-import {TimeElapsed} from './TimeElapsed';
-import {RunDetailsFragment} from './types/RunDetails.types';
 import {RunFragment} from './types/RunFragments.types';
 
-export const timingStringForStatus = (status?: RunStatus) => {
```
Comment (bengotow) [inline js_modules/dagster-ui/packages/ui-core/src/runs/RunConfigDialog.tsx:46]: This is sorta unrelated but I was reading this and cleaned it up a bit. 

- I renamed RunDetails to RunTimingDetails

- The RunConfigDialog was in this file even though it was totally unrelated to the RunDetails. I moved it to a separate file.

File: js_modules/dagster-ui/packages/ui-core/src/runs/RunTimingTags.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import {Box, Popover, Tag} from '@dagster-io/ui-components';
+import * as React from 'react';
+
+import {formatElapsedTime} from '../app/Util';
+import {TimestampDisplay} from '../schedules/TimestampDisplay';
+
+import {RunTimingDetails} from './RunTimingDetails';
+import {RunTimingFragment} from './types/RunTimingDetails.types';
+
+export const RunTimingTags = ({loading, run}: {loading: boolean; run: RunTimingFragment}) => {
```
Comment (bengotow) [inline js_modules/dagster-ui/packages/ui-core/src/runs/RunTimingTags.tsx:10]: Just moved these timing tags to their own file so there's a bit less going on in RunRoot

**Review by hellendag — APPROVED** [review]

**Comment by braunjj** [comment]
This is great! A few minor thoughts:
- How would this look if we made the "n assets" label in the Target column a tag instead of just text? This might look weird but IMO a tag feels more obvious that I can hover over it for actions.
- WTYT about making the two actions "View lineage" and "View downstream lineage".
- What happens when you click one of the actions? Does it navigate to the global graph?
- I like the idea of including these actions in the run actions dropdown as well

File: js_modules/dagster-ui/packages/ui-core/src/assets/AssetsGroupsGlobalGraphRoot.tsx [hunk 3]
```diff
@@ -27,6 +28,21 @@ interface AssetGroupRootParams {
...
+  return `/asset-groups${str}`;
+}
+
+export function globalAssetGraphPathForAssetsAndDescendants(assetKeys: AssetKeyInput[]) {
+  // Note: the max-length of a URL in chrome is 32k characters, and Firefox is 64k characters.
+  // In a perfect world we populate ops query to "asset1*,asset2*" and then select the roots
+  // by passing opNames. If we don't have enough characters to do both, just populate the ops
+  // query. It might still be too long, but we tried.
+  const opsQuery = assetKeys.map((a) => `${tokenForAssetKey(a)}*`).join(', ');
+  const opNames = opsQuery.length > 16000 ? [] : [assetKeys.map(tokenForAssetKey).join(',')];
```
[the remark this answers is not in the document]
Comment (bengotow) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AssetsGroupsGlobalGraphRoot.tsx:42]: Ahh will pull this to a constant and include the division by two - thanks!

**Comment by bengotow** [comment]
Hey @braunjj  I tried the tag appearance in the run target column, but it looks a bit strange because the actual run tags appear just beneath.

What about keeping it a clickable link and adding the actions as well? It's also a bit weird, but I like the visual consistency with op jobs, and when you go to click it you'll notice the other action that is available 🤔

image

**Comment by bengotow** [comment]
Putting the actions into the run dropdowns as well is actually a good bit of work -- going to merge this and make a follow-up for that.

**Pull request merged**